### PR TITLE
AI: Generate PageObject, added types, shell improvement

### PR DESCRIPTION
**Pull request by DavertMik**
* added `askForPageObject` method to generate PageObjects on the fly
* improved AI types
* interactive shell improved to restore history

Screenshot from 2024-06-17 02-47-37

### CODE REVIEW

File: typings/index.d.ts [hunk 1]
```diff
@@ -44,6 +44,40 @@ declare namespace CodeceptJS {
...
   };
 
+  type AiPrompt = {
+    role: string;
+    content: string;
+  }
+
+  type AiConfig = {
+    /** request function to send prompts to AI provider */
+    request: () => Promise<string>,
```
Comment (ductranviet88) [inline typings/index.d.ts]:
```suggestion
    request: (messages: any) => Promise<string>,
```

Comment (ductranviet88) [reply typings/index.d.ts]: it's working with my fixing, appreciate your effort and contribution, Davert.
image

**Review by Arhell — APPROVED** [review]

File: typings/index.d.ts (hunk 1 above)
Comment (kobenguyent) [reply typings/index.d.ts]: thanks for your suggestion!

File: docs/ai.md [hunk 2]
```diff
@@ -315,6 +407,7 @@ ai: {
   prompts: {
     writeStep: (html, input) => [{ role: 'user', content: 'As a test engineer...' }]
     healStep: (html, { step, error, prevSteps }) => [{ role: 'user', content: 'As a test engineer...' }]
+    generatePageObject: (html, extraPrompt = '',  rootLocator = null) => [{ role: 'user', content: 'As a test engineer...' }]
```
Comment (kobenguyent) [inline docs/ai.md:410]: shall we introduce a local var for this `{ role: 'user', content: 'As a test engineer...' }`, as I thought, more prompts would be added

Comment (DavertMik) [reply docs/ai.md:410]: Sorry, I didn't get this.
This is a standard format for prompt, so when you want to override something you should follow the same convention. For instance, you can change the way the PageObjects are generated, you can copy-paste prompt from ai.js and change the way extraProompt is passed in.

Comment (kobenguyent) [reply docs/ai.md:410]: nvm, it's md file. I thought it's the code file.

**Review by kobenguyent — APPROVED** [review]

**Pull request merged**